client/verta/tests/unit_tests/pipeline/test_registered_pipeline.py

```diff
@@ -0,0 +1,300 @@
+        )
+    pipeline_definition = pipeline._to_pipeline_definition()
+    assert pipeline_definition == {
+        "pipeline_version_id": pipeline.id,
+        "graph": graph._to_graph_definition(),
+        "predecessors": graph._to_steps_definition(),
+    }
+
+
+@given(resources=resources())
+def test_to_pipeline_configuration_valid(
+    resources,
+    make_mock_pipeline_graph,
+    make_mock_registered_model_version,
+    make_mock_registered_model,
+) -> None:
+    """Test that a pipeline configuration can be constructed from a
+    RegisteredPipeline object and a valid list of pipeline resources.
+    """
+    mocked_rm = make_mock_registered_model(id=123, name="test_rmv")
+    with patch.object(
+        verta.pipeline.PipelineStep, "_get_registered_model", return_value=mocked_rm
+    ):
+        graph = make_mock_pipeline_graph()
+        step_resources = {step.name: resources for step in graph.steps}
+        pipeline = RegisteredPipeline(
+            graph=graph,
+            registered_model_version=make_mock_registered_model_version(),
+        )
+
+    pipeline_configuration = pipeline._to_pipeline_configuration(
+        pipeline_resources=step_resources
+    )
+    assert pipeline_configuration["pipeline_version_id"] == pipeline.id
+    for graph_step, config_step in zip(graph.steps, pipeline_configuration["steps"]):
+        # All steps are included in the configuration
+        assert graph_step.name == config_step["name"]
+        # All steps in the config have resources
+        assert "resources" in config_step.keys()
+
+
+@given(resources=resources())
+def test_to_pipeline_configuration_invalid_resources(
+    resources,
+    make_mock_pipeline_graph,
+    make_mock_registered_model_version,
+    make_mock_registered_model,
+) -> None:
+    """Test that the expected errors are raised when an invalid pipeline resources
+    are provided.
+
+    Invalid resources include:
+    - a step name not in the pipeline -> ValueError
+    - a step name that is not a string -> TypeError
+    - a step resource that is not a Resources object -> TypeError
+    """
+    mocked_rm = make_mock_registered_model(id=123, name="test_rmv")
+    with patch.object(
+        verta.pipeline.PipelineStep, "_get_registered_model", return_value=mocked_rm
+    ):
+        graph = make_mock_pipeline_graph()
+        step_resources = {step.name: resources for step in graph.steps}
+        step_resources["invalid_step_name"] = resources
+        pipeline = RegisteredPipeline(
+            graph=graph,
+            registered_model_version=make_mock_registered_model_version(),
+        )
+    # step name not in pipeline
+    with pytest.raises(ValueError) as err:
+        pipeline._to_pipeline_configuration(pipeline_resources=step_resources)
+        assert (
+                str(err.value)
+                == "pipeline_resources contains resources for a step not in the "
+                   "pipeline: 'invalid_step_name'"
+        )
```

Comment:
These assertions need to be moved _outside_ the `with` block. Written here, they aren't even being run at all. Does that make sense?

Reply:
Ah, I see.  Once pytest catches the exception, it doesn't continue with any more code in that block.  Silly error on my part.  Fixed